wormhole/file_transport.go

```diff
@@ -188,21 +192,26 @@ func (t *fileTransport) connectViaRelay(otherTransit *transitMsg) (net.Conn, err
 
 	var count int
 
+	fmt.Println(otherTransit.HintsV1)
 	for _, relay := range otherTransit.HintsV1 {
 		if relay.Type == "relay-v1" {
 			for _, endpoint := range relay.Hints {
-				var addr string
+				var relayUrl *url.URL
 				switch endpoint.Type {
 				case "direct-tcp-v1":
-					addr = net.JoinHostPort(endpoint.Hostname, strconv.Itoa(endpoint.Port))
+					relayUrl = &url.URL{
+						Scheme: "tcp",
+						Host:   net.JoinHostPort(endpoint.Hostname, strconv.Itoa(endpoint.Port)),
+					}
 				case "websocket-v1":
-					addr = endpoint.Url
+					relayUrl, _ = url.Parse(endpoint.Url)
```

Comment:
I don't think we can ignore that error here. Because `endpoints` is input from the other side, and thus may be malformed.

Reply:
Added. However if error is received, now terminates download, even if next url is correct. Could find out how to cancel one and move to another url try (Tried, but always hangs on channel waiting).

Reply:
I think you could just `continue` in the loop

Reply:
Thanks. Added. Not it skips incorrect url and moves to working.